component/module/git/internal/vcs/auth/auth.go

```diff
@@ -0,0 +1,67 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/go-git/go-git/v5/plumbing/transport/ssh"
+	"github.com/grafana/agent/pkg/river/rivertypes"
+)
+
+type GitAuthConfig struct {
+	BasicAuth *BasicAuth `river:"basic_auth,block,optional"`
+	SSHKey    *SSHKey    `river:"ssh_key,block,optional"`
+}
+
+// Convert converts HTTPClientConfig to the native Prometheus type. If h is
+// nil, the default client config is returned.
+func (h *GitAuthConfig) Convert() transport.AuthMethod {
+	if h == nil {
+		return nil
+	}
+
+	if h.BasicAuth != nil {
+		return h.BasicAuth.Convert()
+	}
+
+	if h.SSHKey != nil {
+		key, _ := h.SSHKey.Convert()
+		return key
+	}
+	return nil
+}
+
+type BasicAuth struct {
+	Username string            `river:"username,attr,optional"`
+	Password rivertypes.Secret `river:"password,attr,optional"`
+}
+
+// Convert converts our type to the native prometheus type
+func (b *BasicAuth) Convert() (t transport.AuthMethod) {
+	if b == nil {
+		return nil
+	}
+	return &http.BasicAuth{
+		Username: b.Username,
+		Password: string(b.Password),
+	}
+}
+
+type SSHKey struct {
```

Comment:
In the context of SSH keys, I don't think these are all optional; at least the key file would be required, right?

Reply:
I marked these optional to match the BasicAuth block. I didn't experiment with what happens marking these as required, but I assumed if the block is optional all it's attributes also must be optional.

Reply:
It's valid for an optional block to have required attributes; it means those attributes are only required if the block is specified in the config file.

Reply:
I just looked at how auth is done for remote.vault and saw required attributes in optional blocks. Added a commit to address this.

I think I've fixed everything now. Thank you for your patience with me reviewing this!